packages/node/src/integrations/tracing/nest.ts

```diff
@@ -44,7 +54,150 @@ interface MinimalNestJsApp {
 
 const INTEGRATION_NAME = 'Nest';
 
-export const instrumentNest = generateInstrumentOnce(INTEGRATION_NAME, () => new NestInstrumentation());
+const supportedVersions = ['>=8.0.0 <11'];
+
+const sentryPatched = 'sentryPatched';
+
+/**
+ * Represents an injectable target class in NestJS.
+ */
+interface InjectableTarget {
+  name: string;
+  sentryPatched?: boolean;
+  prototype: {
+    use?: (req: unknown, res: unknown, next: (error?: Error | unknown) => void) => void;
+  };
+}
+
+/**
+ * Helper checking if a target is already patched.
+ */
+function isPatched(target: InjectableTarget): boolean {
```

Comment:
Why do we need this in addition to `isWrapped`?

Reply:
described below, will add a comment